contracts/test/unit/entry/SystemEntry.t.sol

```diff
@@ -0,0 +1,135 @@
+    MockPCVDepositV3 private deposit2;
+    // test Oracles
+    MockOracle private oracle1;
+    MockOracle private oracle2;
+
+    // mocked DynamicVoltSystemOracle behavior to prevent reverts
+    uint256 private liquidReserves;
+
+    function updateActualRate(uint256 _liquidReserves) external {
+        liquidReserves = _liquidReserves;
+    }
+
+    function setUp() public {
+        // volt system
+        core = CoreV2(address(getCoreV2()));
+        pcvOracle = new PCVOracle(address(core));
+        entry = new SystemEntry(address(core));
+
+        // mock utils
+        oracle1 = new MockOracle();
+        token1 = new MockERC20();
+        deposit1 = new MockPCVDepositV3(address(core), address(token1));
+
+        // grant role
+        vm.startPrank(addresses.governorAddress);
+        core.grantLocker(address(entry));
+        core.grantLocker(address(pcvOracle));
+        core.grantLocker(address(deposit1));
+        entry.setPCVOracle(address(pcvOracle));
+        vm.stopPrank();
+
+        // setup pcv oracle
+        address[] memory venues = new address[](1);
+        venues[0] = address(deposit1);
+        address[] memory oracles = new address[](1);
+        oracles[0] = address(oracle1);
+        bool[] memory isLiquid = new bool[](1);
+        isLiquid[0] = true;
+        vm.prank(addresses.governorAddress);
+        pcvOracle.addVenues(venues, oracles, isLiquid);
+    }
+
+    function testSetup() public {
+        assertEq(address(entry.core()), address(core));
+        assertTrue(!entry.paused());
+    }
+
+    // --------------- Happy paths --------------------------
+
+    function testAccrue() public {
+        vm.expectCall(address(deposit1), abi.encodeWithSignature("accrue()"));
+        entry.accrue(address(deposit1));
+    }
+
+    function testDeposit() public {
+        vm.expectCall(address(deposit1), abi.encodeWithSignature("deposit()"));
+        entry.deposit(address(deposit1));
+    }
+
+    function testHarvest() public {
+        vm.expectCall(address(deposit1), abi.encodeWithSignature("harvest()"));
+        entry.harvest(address(deposit1));
+    }
+
+    // --------------- Paused state reverts -----------------
+
+    function testAccrueRevertsIfPaused() public {
+        vm.prank(addresses.governorAddress);
+        entry.pause();
+
+        vm.expectRevert("Pausable: paused");
+        entry.accrue(address(deposit1));
+    }
+
+    function testDepositRevertsIfPaused() public {
+        vm.prank(addresses.governorAddress);
+        entry.pause();
+
+        vm.expectRevert("Pausable: paused");
+        entry.deposit(address(deposit1));
+    }
+
+    function testHarvestRevertsIfPaused() public {
+        vm.prank(addresses.governorAddress);
+        entry.pause();
+
+        vm.expectRevert("Pausable: paused");
+        entry.harvest(address(deposit1));
+    }
+
+    // --------------- Invalid deposit reverts --------------
+
+    function testAccrueRevertsIfWrongDeposit() public {
```

Comment:
would be great to see a test where pcvOracle is not set in core, and then you can call deposit, harvest or accrue on any pcv deposit.

Reply:
let's add this in the integration tests for system v2 because it's mostly testing interactions